prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -562,17 +566,24 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
         private const val TIMEOUT = 120_000
 
         /** Our default Http Client, with an optional SSL context, and optional auth token */
-        private fun createDefaultHttpClient(jks: UserJksCredential?, accessToken: String?): HttpClient {
+        private fun createDefaultHttpClient(
+            jks: UserJksCredential?,
+            accessToken: String?,
+                                            restTransportInfo: RESTTransportType?,
+        ): HttpClient {
             return HttpClient(Apache) {
                 // installs logging into the call to post to the server
                 install(Logging) {
                     logger = io.ktor.client.plugins.logging.Logger.Companion.SIMPLE
                     level = LogLevel.INFO // LogLevel.INFO for prod, LogLevel.ALL to view full request
                 }
+
                 // not using Bearer Auth handler due to refresh token behavior
                 accessToken?.let {
+                    // Default to set Bearer prefix
+                    val authPrefix = restTransportInfo!!.headers["BearerToken"] ?: "Bearer"
```

Comment:
I think it might be better to extract this logic into its own function and have unit tests for it. To make sure this doesn't break again.

Reply:
Ok, I will do that.